server/jetty/src/main/java/io/deephaven/server/jetty/JettyConfig.java

```diff
@@ -68,15 +77,60 @@ public int port() {
     }
 
     /**
-     * Include websockets. Defaults to {@value DEFAULT_WITH_WEBSOCKETS}.
+     * Include websockets.
      */
-    @Default
-    public boolean websockets() {
-        return DEFAULT_WITH_WEBSOCKETS;
+    @Nullable
+    public abstract Boolean websockets();
+
+    /**
+     * Include HTTP/1.1.
+     */
+    @Nullable
+    public abstract Boolean http1();
+
+    /**
+     * Returns {@link #websockets()} if explicitly set. If {@link #proxyHint()} is {@code true}, returns {@code false}.
+     * Otherwise, defaults to {@code true} when {@link #ssl()} is empty, and {@code false} when {@link #ssl()} is
+     * present.
+     */
+    public final boolean websocketsOrDefault() {
+        final Boolean websockets = websockets();
+        if (websockets != null) {
+            return websockets;
+        }
+        if (Boolean.TRUE.equals(proxyHint())) {
+            return false;
+        }
+        return ssl().isEmpty();
+    }
+
+    /**
+     * Returns {@link #http1()} if explicitly set. If {@link #proxyHint()} is {@code true}, returns {@code false}.
+     * Otherwise, defaults to {@code true} when {@link #ssl()} is empty, and {@code false} when {@link #ssl()} is
+     * present.
+     */
+    public final boolean http1OrDefault() {
+        final Boolean http1 = http1();
+        if (http1 != null) {
+            return http1;
+        }
+        if (Boolean.TRUE.equals(proxyHint())) {
+            return false;
+        }
+        return ssl().isEmpty();
+    }
+
+    @Value.Check
+    final void checkWebsocketsNeedsHttp1() {
+        if (websocketsOrDefault() && !http1OrDefault()) {
+            throw new IllegalArgumentException("Jetty configuration includes websockets but not HTTP1.1");
```

Comment:
It is my understanding that there is no reason that websockets _can't_ work on h2 - for example (as we have discussed elsewhere) if the server is behind multiple proxies, and one doesn't support h2, we would want to encourage websockets.

Example:
```mermaid
graph TD
    C[deephaven-server-jetty-app]
    B[Proxy #1]
    A[Proxy #2]
    D[plain h2 grpc client]
    E["public internet"]
    A-->|http/1.1|B
    D-->|h2|C
    B-->|h2|C
    E-->A
```

Reply:
Ah yes - I think this fact was established after I created this PR. I'll remove this improper check.